sphinxcontrib/confluencebuilder/storage/translator.py

```diff
@@ -1145,13 +1150,9 @@ def visit_table(self, node):
         table_classes = node.get('classes', [])
         attribs = {}
 
-        # For v2 editor, if we have given explicit widths for columns in the
-        # table (e.g. CSV table), we need to apply a data table width or the
-        # editor will ignore the column-specific widths. If widths are
-        # detected, apply the default table width observed when using the v2
-        # editor.
-        if self.v2 and 'colwidths-given' in table_classes:
-            attribs['data-table-width'] = CONFLUENCE_DEFAULT_V2_TABLE_WIDTH
+        # Apply the default table width when using the v2 editor.
+        if self.v2:
+            attribs['data-table-width'] = self._v2_default_table_width
```

Comment:
Would it be possible to do something more like this:

```
        # Apply a user-provided default table width when using the v2 editor.
        if self.v2 and self._v2_default_table_width:
            attribs['data-table-width'] = self._v2_default_table_width
        # For v2 editor, if we have given explicit widths for columns in the
        # table (e.g. CSV table), we need to apply a data table width or the
        # editor will ignore the column-specific widths. If widths are
        # detected, apply the default table width observed when using the v2
        # editor.
        elif self.v2 and 'colwidths-given' in table_classes:
            attribs['data-table-width'] = CONFLUENCE_DEFAULT_V2_TABLE_WIDTH
```

(sorry for not being clear in the previous comments)

Reply:
This would counteract my intention of the whole change. The idea was, even if colwidths-given is true to be able to set a table width. On our company cloud hosted confluence this works fine to set both. Otherwise some of our tables are way to narrow.

If colwidths are not set then the table look weird because the columns are not streched in a sense that the full header text is in one line or in any way readable.

Reply:
On the other hand can _v2_default_table_width be not set? Doesn't the config code set it to CONFLUENCE_DEFAULT_V2_TABLE_WIDTH if it is not set in the config?
If so it would mean the ```elif``` is never used.

Reply:
In `defaults.py`, we could change:

```
    if conf.confluence_default_table_width is None:
        conf.confluence_default_table_width = CONFLUENCE_DEFAULT_V2_TABLE_WIDTH
```

To be:

```
    if conf.confluence_default_table_width is not None:
        table_width, twu = extract_length(str(self.confluence_default_table_width))
        if table_width is not None:
            self.confluence_default_table_width = convert_length(table_width, twu, pct=False)
```

And in `storage/translator.py`, drop `self._v2_default_table_width` and directly use `confluence_default_table_width`.